### ISSUE TITLE
nixos/dendrite: private_key must be set and private_key can not be automatically generated

### ISSUE THREAD

**Issue by lions-tech** [NixOS/nixpkgs#225845]
### Describe the bug
When setting up the dendrite matrix server, one must specify a `settings.global.private_key` otherwise it will not build (`error: The option 'services.dendrite.settings.global.private_key' is used but not defined`). That is because the `private_key` option does not have a default. Additionally the `private_key` must be generated manually after boot, if the service module is not overriden. IMO that is very impractical.

### Steps To Reproduce
1. Setup a minimal flake with a minimal dendrite config as follows:
```nix
services.dendrite = {
  enable = true;
  settings.global.server_name = "localhost";
};
```
2. Try to build the flake

### Expected behavior
Should build without needing to specify a `private_key` explicitely and should generate a `private_key` on startup.

### Additional context
Here is how I implemented the private_key generation locally:
```nix
systemd.services.dendrite.serviceConfig.ExecStartPre = lib.mkForce [
  ''${pkgs.bash}/bin/bash -c "if [ ! -f ${private_key} ]; then ${pkgs.coreutils}/bin/mkdir -p /etc/dendrite/keys; ${pkgs.dendrite}/bin/generate-keys --private-key ${private_key}; ${pkgs.coreutils}/bin/chmod 666 ${private_key}; fi;"''
  ''${pkgs.envsubst}/bin/envsubst -i ${configurationYaml} -o ${configPath}''
];
```

### Notify maintainers
@ma27 @fadenb @mguentner @ekleog @ralith @dandellion @sumnerevans

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

This the ouptut on the machine on which I want to build:
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.103, NixOS, 22.11 (Raccoon), 22.11pre431886.eac7da7b519-esg`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.11.1`
 - channels(root): `"vas"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/vas/nixpkgs`
```

**Comment by ocfox** [comment]
I think that this is not a security behavior to generate a private key when build time.

**Comment by lions-tech** [comment]
> I think that this is not a security behavior to generate a private key when build time.

I agree, for that reason I would generate them before starting the service (see my example).
